lib/pages/chat/chat_event_list.dart

```diff
@@ -203,6 +211,41 @@ class ChatEventList extends StatelessWidget {
     );
   }
 
+  void _cleanCancelTokenMap(Event event) {
+    if (event.status == EventStatus.sent &&
+        controller.mediaCancelTokenMapNotifier.value[event.body] != null) {
+      controller.mediaCancelTokenMapNotifier.value.remove(event.body);
+    }
+  }
+
+  Future<void> _cancelSending({
```

Comment:
it is a stateless widget, why we should keep logic here?

Reply:
It is here because:
- it has a strong relationship with the Message widget, so I don't think we should use a callback that needs to propagate back to the controller. I'm afraid it might reduce the code's readability.
- it needs the event used in the Message widget
- it does not need a `setState()` call, so for me no need to replace it with a `StatefulWidget`

WDYT ?